src/UnigramWordPredictor.java

```diff
@@ -52,60 +54,91 @@ public void train(Scanner scanner) {
     List<String> trainingWords = tokenizer.tokenize(scanner);
 
     // TODO: Convert the trainingWords into neighborMap here
+    //Map<String, List<String>> neighborMap = new HashMap<>();
+    this.neighborMap = new HashMap<>();
+
+    for (int i = 0; i < trainingWords.size() - 1; i++) {
+
+      String word = trainingWords.get(i);
+      String nextWord = trainingWords.get(i + 1);
+
+      if (!neighborMap.containsKey(word)) {
+        List<String> mapValue = new ArrayList<>();
+
+        mapValue.add(nextWord);
+
+        neighborMap.put(word, mapValue);
+      } else {
+        neighborMap.get(word).add(nextWord);
+      }
```

Comment:
Nice! As an alternative, can you think about how you could write this using `putIfAbsent` or `computeIfAbsent` instead?